tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1060,7 +1060,7 @@ def test_single_sign_on_ldap_ad_server(
         else:
             curl_command = f'curl -k -u : --negotiate {default_sat.url}/users/extlogin/'
         result = run_command(curl_command)
```

Comment:
`run_command` was the function written to wrap ssh.command with assertion. I think needs to updated. Let me know if you wanted to do that refactor or I can work on it remove those things.

Reply:
I've talked to @JacobCallahan about the frequency of the pattern of asserting the RC, and that we should have a separate method - or `check` kwarg` - for the execute method.  This would follow python's `subprocess.run' pattern.

It won't be part of this PR, because its a change that can easily be made separate from the rest of the refactoring efforts. After we've completed this PR and merging unstable into master, someone can pick up that effort.